### PR TITLE
Improve the Kafka + schema registry documentation for Confluent Schema Registry

### LINKED ISSUE #19350: Improve guide for Kafka + Avro projects
Hi all,
following this guide https://quarkus.io/guides/kafka-schema-registry-avro I noticed that it's necessary to add something in the "Creating the Maven Project" section.

If you use Confluent Schema Registry, you don’t need the `quarkus-apicurio-registry-avro`, but if you use Confluent's `kafka-avro-serializer`, the mechanins to auto-generate classes from avro schemas won't work since the `quarkus-avro` dependency will be missing.

In the snippet in the tip maybe you should add the `quarkus-avro` dependency, or at least add a phrases to warn the reader.

Thank you,
Mauro

### PR DESCRIPTION
**Pull request by Ladicek**
Fixes #19350.

### CODE REVIEW

**Review by ozangunalp — APPROVED** [review]

**Comment by Ladicek** [comment]
I have added quite a bit more details about Confluent Schema Registry.

File: docs/src/main/asciidoc/kafka-schema-registry-avro.adoc [hunk 1]
```diff
@@ -384,7 +420,7 @@ java -Dmp.messaging.connector.smallrye-kafka.apicurio.registry.url=http://localh
 ----
 
 NOTE: By default, the application tries to connect to a Kafka broker listening at `localhost:9092`.
-You can configure the bootstrap server using: `java -Dkafka.bootstrap.servers=... -jar target/quarkus-app/quarkus-run.jar`
+You can configure the bootstrap server using: `java -Dkafka.bootstrap.servers=\... -jar target/quarkus-app/quarkus-run.jar`
```
Comment (gsmet) [inline docs/src/main/asciidoc/kafka-schema-registry-avro.adoc:423]: Why did you add a `\`? AFAICS, the current doc is properly rendered on the website.

**Review by ozangunalp — APPROVED** [review]
Nice addition! The guide works as described with the Confluent Schema Registry.

**Review by gsmet — APPROVED** [review]
